packages/terra-search-field/src/SearchField.jsx

```diff
@@ -183,36 +213,59 @@ class SearchField extends React.Component {
       inputAttributes,
       ...customProps
     } = this.props;
+
+    const clearButtonVisible = !this.state.searchText ? 'input-clear' : undefined;
+
+    const inputClassNames = cx([
+      'input',
+      clearButtonVisible,
+    ]);
+
     const searchFieldClassNames = cx([
       'searchfield',
       { block: isBlock },
       customProps.className,
     ]);
 
-    const valueTemp = {};
+    const inputText = this.context.intl.formatMessage({ id: 'Terra.searchField.search' });
+    const buttonText = this.context.intl.formatMessage({ id: 'Terra.searchField.submit-search' });
+    const clearText = this.context.intl.formatMessage({ id: 'Terra.searchField.clear' });
+    const additionalInputAttributes = Object.assign({ 'aria-label': inputText }, inputAttributes);
+    const clearIcon = <span className={cx('clear-icon')} />;
+
     if (value !== undefined) {
-      valueTemp.value = value;
+      additionalInputAttributes.value = value;
     } else {
-      valueTemp.defaultValue = defaultValue;
+      additionalInputAttributes.defaultValue = defaultValue;
     }
 
-    const inputText = this.context.intl.formatMessage({ id: 'Terra.searchField.search' });
-    const buttonText = this.context.intl.formatMessage({ id: 'Terra.searchField.submit-search' });
-    const additionalInputAttributes = Object.assign({ 'aria-label': inputText }, inputAttributes, valueTemp);
+    const clearButton = this.state.searchText
+      ? (
+        <Button
+          className={cx('clear')}
+          onClick={this.handleClear}
+          text={clearText}
+          variant="utility"
+          icon={clearIcon}
+          isIconOnly
+        />
+      )
+      : undefined;
 
     return (
       <div {...customProps} className={searchFieldClassNames}>
         <Input
-          className={cx('input')}
+          className={inputClassNames}
           type="search"
           placeholder={placeholder}
           onChange={this.handleTextChange}
           disabled={isDisabled}
           aria-disabled={isDisabled}
           onKeyDown={this.handleKeyDown}
-          refCallback={inputRefCallback}
+          refCallback={inputRefCallback || this.setInputRef}
```

Comment:
You should just be passing in this.setInputRef here, it's already wrapping inputRefCallback. Doing it this way if the consumer passes in an inputRefCallback setInputRef won't get passed in.

Reply:
Updated: [6f3c0ac](https://github.com/cerner/terra-core/pull/2317/commits/6f3c0ac8a332bae73c766a02470db83de37aec28). Also updated the jest test associated with the callback!